### PR TITLE
PATCH: patient linking

### PR DESCRIPTION
**Pull request by Orta21**
Part of cs-2038

Issues:

- https://linear.app/metriport/issue/CS-2038

### Dependencies

- Upstream: none
- Downstream: none

### Description

- Optimizing patient linking to properly validate links coming in

### Testing

- Local
  - [x] Local script going through list of links
  - [x] unit tests
- Production
  - [ ] Monitor production to ensure links are being flagged accordingly

### Release Plan

- :warning: Points to `master`
- [ ] Merge this

## Summary by CodeRabbit

* **New Features**
  * Stronger contact matching using normalized phone and email.
  * Cross-link validation now accepts last-name matches when paired with first name, exact DOB, or address.

* **Bug Fixes**
  * Address matching more resilient to apartment/unit formatting and reordered lines.
  * Phone formats (e.g., (555)123-4567 vs 5551234567) normalized to avoid false mismatches.
  * Business-rule failure messages now include "No Contact Match" when applicable.

* **Tests**
  * Expanded tests covering address interoperability, cross-link scenarios, and contact-edge cases.

### CODE REVIEW

File: packages/core/src/mpi/__tests__/filter-patients.test.ts [hunk 1]
```diff
@@ -188,6 +188,102 @@ describe("evaluatePatientMatch", () => {
...
+            zip: "10001",
+          },
+        ],
+      };
+
+      const result = await evaluatePatientMatch(patient1, patient2);
+
+      expect(result.scores.address).toBe(2);
+    });
+
```
[the remark this answers is not in the document]
Comment (Orta21) [reply packages/core/src/mpi/__tests__/filter-patients.test.ts:255]: The chances of a person having the same demographics and living in the same building are not very likely

File: packages/core/src/mpi/filter-patients/cross-validate-links.ts [hunk 2]
```diff
@@ -16,6 +17,12 @@ export function crossValidateInvalidLinks(
 
 function isLinkValidByAssociation(invalidLink: PatientData, validLinks: PatientData[]): boolean {
   for (const validLink of validLinks) {
+    const hasLastNameMatch = calculateLastNameScore(invalidLink, validLink);
+
+    if (hasLastNameMatch) {
+      return true;
+    }
```
[the remark this answers is not in the document]
Comment (Orta21) [reply packages/core/src/mpi/filter-patients/cross-validate-links.ts]: fixed
